app/screens/receive-bitcoin-screen/use-receive-bitcoin.ts

```diff
@@ -0,0 +1,249 @@
+import { gql } from "@apollo/client"
+import {
+  useLnInvoiceCreateMutation,
+  useLnNoAmountInvoiceCreateMutation,
+  useLnUsdInvoiceCreateMutation,
+  useOnChainAddressCurrentMutation,
+  WalletCurrency,
+} from "@app/graphql/generated"
+import { useLnUpdateHashPaid } from "@app/graphql/ln-update-context"
+import { PaymentAmount } from "@app/types/amounts"
+import { WalletDescriptor } from "@app/types/wallets"
+import { useCallback, useEffect, useMemo, useState } from "react"
+import { ConvertPaymentAmount } from "../send-bitcoin-screen/payment-details"
+import { createInvoiceDetails, CreateInvoiceDetailsParams } from "./invoices"
+import { InvoiceType } from "./invoices/index.types"
+import {
+  ErrorType,
+  InvoiceState,
+  ReceiveBitcoinState,
+  UseReceiveBitcoinParams,
+  UseReceiveBitcoinResult,
+} from "./use-receive-bitcoin.types"
+
+gql`
+  mutation lnNoAmountInvoiceCreate($input: LnNoAmountInvoiceCreateInput!) {
+    lnNoAmountInvoiceCreate(input: $input) {
+      errors {
+        message
+      }
+      invoice {
+        paymentHash
+        paymentRequest
+        paymentSecret
+      }
+    }
+  }
+
+  mutation lnInvoiceCreate($input: LnInvoiceCreateInput!) {
+    lnInvoiceCreate(input: $input) {
+      errors {
+        message
+      }
+      invoice {
+        paymentHash
+        paymentRequest
+        paymentSecret
+        satoshis
+      }
+    }
+  }
+
+  mutation onChainAddressCurrent($input: OnChainAddressCurrentInput!) {
+    onChainAddressCurrent(input: $input) {
+      errors {
+        message
+      }
+      address
+    }
+  }
+
+  mutation lnUsdInvoiceCreate($input: LnUsdInvoiceCreateInput!) {
+    lnUsdInvoiceCreate(input: $input) {
+      errors {
+        message
+      }
+      invoice {
+        paymentHash
+        paymentRequest
+        paymentSecret
+        satoshis
+      }
+    }
+  }
+`
+
+export const useReceiveBitcoin = ({
```

Comment:
when I see `useReceiveBitcoin`, I'm expecting it's a bitcoin only wallet flow. I think this hook can also be used to receive stablesats, so maybe a better name would just be `useReceive`, or `useReceiveFunds`. something along those lines that will convey that it's not only for bitcoin wallets

Reply:
Sounds good!